From 9969511bc8c24fdbf526193a4585a2db60181787 Mon Sep 17 00:00:00 2001
From: Brian Wood <brian.j.wood@intel.com>
Date: Thu, 26 Jul 2018 16:34:46 -0700
Subject: [PATCH 1753/2345] Fix for memory leak in hdmi write

There is a memory leak condition in intel_hdmi_hdcp_write
where a buffer is kzalloc'd but there is no call to kfree
the buffer in the code calls downstream. This commit fixes
the issue.

Signed-off-by: Brian Wood <brian.j.wood@intel.com>
---
 drivers/gpu/drm/i915/intel_hdmi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/i915/intel_hdmi.c b/drivers/gpu/drm/i915/intel_hdmi.c
index 60fd172..8e60071 100644
--- a/drivers/gpu/drm/i915/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/intel_hdmi.c
@@ -924,6 +924,7 @@ static int intel_hdmi_hdcp_write(struct intel_digital_port *intel_dig_port,
 	msg.buf = write_buf;
 
 	ret = i2c_transfer(adapter, &msg, 1);
+	kfree(write_buf);
 	if (ret == 1)
 		return 0;
 	return ret >= 0 ? -EIO : ret;
-- 
2.7.4

